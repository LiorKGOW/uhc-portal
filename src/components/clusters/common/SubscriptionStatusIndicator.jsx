import React from 'react';
import PropTypes from 'prop-types';
import get from 'lodash/get';
import { Popover, PopoverPosition, Button } from '@patternfly/react-core';
import { CheckCircleIcon, WarningTriangleIcon, UnknownIcon } from '@patternfly/react-icons';
// eslint-disable-next-line camelcase
import { global_success_color_100, global_warning_color_100 } from '@patternfly/react-tokens';

function SubscriptionStatusIndicator({ cluster }) {
  const managed = get(cluster, 'managed');
  const entitlementStatus = get(cluster, 'subscription.entitlement_status');

  switch (entitlementStatus) {
    case 'Ok':
      return (
        <React.Fragment>
          <CheckCircleIcon color={global_success_color_100.value} />
          {' '}
          Subscribed
        </React.Fragment>
      );
    case 'Valid':
      return (
        <React.Fragment>
          <CheckCircleIcon color={global_success_color_100.value} />
          {' '}
          Valid
        </React.Fragment>
      );
    case 'NotSet':
      return (
        <Popover
          position={PopoverPosition.top}
          bodyContent="This cluster does not have a subscription attached."
          aria-label="Not Subscribed"
        >
          <Button variant="link" isInline icon={<WarningTriangleIcon color={global_warning_color_100.value} />}>
            Not Subscribed
          </Button>
        </Popover>
      );
    case 'Overcommitted':
      return (
        <Popover
          position={PopoverPosition.top}
          bodyContent="This cluster is consuming more resources than it is entitled to."
          aria-label="Insufficient"
        >
          <Button variant="link" isInline icon={<WarningTriangleIcon color={global_warning_color_100.value} />}>
            Insufficient
          </Button>
        </Popover>
      );
    case 'InconsistentServices':
      return (
        <Popover
          position={PopoverPosition.top}
          bodyContent="This cluster is attached to subscriptions with different support levels."
          aria-label="Invalid"
        >
          <Button variant="link" isInline icon={<WarningTriangleIcon color={global_warning_color_100.value} />}>
            Invalid
          </Button>
        </Popover>
      );
    default:
      if (managed) {
        return (
          <React.Fragment>
            <CheckCircleIcon color={global_success_color_100.value} />
            {' '}
            Subscribed
          </React.Fragment>
        );
      }
      return (
        <Popover
          position={PopoverPosition.top}
          bodyContent="Subscription information for this cluster is not available yet. It may take up to 12 hours for this information to become available."
          aria-label="Unknown"
        >
          <Button variant="link" isInline icon={<UnknownIcon color="black" />}>
            Unknown
          </Button>
        </Popover>
      );
  }
}

SubscriptionStatusIndicator.propTypes = {
  cluster: PropTypes.shape({
    managed: PropTypes.bool,
    subscription: PropTypes.shape({
      entitlement_status: PropTypes.string,
    }),
  }),
};

export default SubscriptionStatusIndicator;
