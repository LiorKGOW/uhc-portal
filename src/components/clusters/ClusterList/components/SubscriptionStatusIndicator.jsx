import React from 'react';
import PropTypes from 'prop-types';
import get from 'lodash/get';
import { Popover, PopoverPosition, Button } from '@patternfly/react-core';
import { CheckCircleIcon, ExclamationTriangleIcon } from '@patternfly/react-icons';
// eslint-disable-next-line camelcase
import { global_success_color_100, global_warning_color_100 } from '@patternfly/react-tokens';
import { entitlementStatuses } from '../../../../common/subscriptionTypes';
import getClusterEvaluationExpiresInDays from '../../../../common/getClusterEvaluationExpiresInDays';

function SubscriptionStatusIndicator({ cluster }) {
  const managed = get(cluster, 'managed');
  const entitlementStatus = get(cluster, 'subscription.entitlement_status');
  const evaluationExpiresStr = getClusterEvaluationExpiresInDays(cluster);

  switch (entitlementStatus) {
    case entitlementStatuses.OK:
      return (
        <>
          <CheckCircleIcon color={global_success_color_100.value} />
          {' '}
          Subscribed
        </>
      );
    case entitlementStatuses.OVERCOMMITTED:
      return (
        <Popover
          position={PopoverPosition.top}
          bodyContent="This cluster is consuming more resources than it is entitled to."
          aria-label="Insufficient"
        >
          <Button variant="link" isInline icon={<ExclamationTriangleIcon color={global_warning_color_100.value} />}>
            Insufficient
          </Button>
        </Popover>
      );
    case entitlementStatuses.INCONSISTENT_SERVICES:
      return (
        <Popover
          position={PopoverPosition.top}
          bodyContent="This cluster is attached to subscriptions with different support levels."
          aria-label="Invalid"
        >
          <Button variant="link" isInline icon={<ExclamationTriangleIcon color={global_warning_color_100.value} />}>
            Invalid
          </Button>
        </Popover>
      );
    case entitlementStatuses.SIXTY_DAY_EVALUATION:
      return (
        <Popover
          position={PopoverPosition.top}
          bodyContent={`${evaluationExpiresStr} remaining in the evaluation period.`}
          aria-label="Six Day Evaluation"
        >
          <Button variant="link" isInline icon={<ExclamationTriangleIcon color={global_warning_color_100.value} />}>
            60-day evaluation
          </Button>
        </Popover>
      );
    default:
      if (managed) {
        return (
          <>
            <CheckCircleIcon color={global_success_color_100.value} />
            {' '}
            Subscribed
          </>
        );
      }
      return (
        <Popover
          position={PopoverPosition.top}
          bodyContent="This cluster does not have a subscription attached."
          aria-label="Not Subscribed"
        >
          <Button variant="link" isInline icon={<ExclamationTriangleIcon color={global_warning_color_100.value} />}>
            Not subscribed
          </Button>
        </Popover>
      );
  }
}

SubscriptionStatusIndicator.propTypes = {
  cluster: PropTypes.shape({
    managed: PropTypes.bool,
    subscription: PropTypes.shape({
      entitlement_status: PropTypes.string,
    }),
    creation_timestamp: PropTypes.string,
  }),
};

export default SubscriptionStatusIndicator;
