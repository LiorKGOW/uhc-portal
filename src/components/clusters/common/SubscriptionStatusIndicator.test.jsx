import React from 'react';
import { shallow } from 'enzyme';
import SubscriptionStatusIndicator from './SubscriptionStatusIndicator';

describe('<SubscriptionStatusIndicator />', () => {
  it('should not crash when the cluster has no subscription info', () => {
    const wrapper = shallow(<SubscriptionStatusIndicator subscriptionInfo={undefined} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render when subscription is Ok', () => {
    const wrapper = shallow(<SubscriptionStatusIndicator subscriptionInfo={{ entitlement_status: 'Ok' }} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render when subscription is Valid', () => {
    const wrapper = shallow(<SubscriptionStatusIndicator subscriptionInfo={{ entitlement_status: 'Valid' }} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render when subscription is NotSet', () => {
    const wrapper = shallow(<SubscriptionStatusIndicator subscriptionInfo={{ entitlement_status: 'NotSet' }} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render when subscription is Overcommitted', () => {
    const wrapper = shallow(<SubscriptionStatusIndicator subscriptionInfo={{ entitlement_status: 'Overcommitted' }} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render when subscription is InconsistentServices', () => {
    const wrapper = shallow(<SubscriptionStatusIndicator subscriptionInfo={{ entitlement_status: 'InconsistentServices' }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
