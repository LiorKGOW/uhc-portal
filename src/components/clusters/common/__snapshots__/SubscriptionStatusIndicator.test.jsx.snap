// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubscriptionStatusIndicator /> should not crash when the cluster has no subscription info 1`] = `
<Popover
  appendTo={[Function]}
  aria-label="Unknown"
  bodyContent="Subscription information for this cluster is not available yet. It may take up to 12 hours for this information to become available."
  className=""
  closeBtnAriaLabel="Close"
  enableFlip={true}
  footerContent={null}
  headerContent={null}
  hideOnOutsideClick={true}
  isVisible={null}
  maxWidth="calc(2rem + 18.75rem)"
  onHidden={[Function]}
  onHide={[Function]}
  onMount={[Function]}
  onShow={[Function]}
  onShown={[Function]}
  position="top"
  shouldClose={[Function]}
  zIndex={9999}
>
  <Button
    icon={
      <UnknownIcon
        color="black"
        noVerticalAlign={false}
        size="sm"
        title={null}
      />
    }
    isInline={true}
    variant="link"
  >
    Unknown
  </Button>
</Popover>
`;

exports[`<SubscriptionStatusIndicator /> should render when subscription is InconsistentServices 1`] = `
<Popover
  appendTo={[Function]}
  aria-label="Invalid"
  bodyContent="This cluster is attached to subscriptions with different support levels."
  className=""
  closeBtnAriaLabel="Close"
  enableFlip={true}
  footerContent={null}
  headerContent={null}
  hideOnOutsideClick={true}
  isVisible={null}
  maxWidth="calc(2rem + 18.75rem)"
  onHidden={[Function]}
  onHide={[Function]}
  onMount={[Function]}
  onShow={[Function]}
  onShown={[Function]}
  position="top"
  shouldClose={[Function]}
  zIndex={9999}
>
  <Button
    icon={
      <WarningTriangleIcon
        color="#f0ab00"
        noVerticalAlign={false}
        size="sm"
        title={null}
      />
    }
    isInline={true}
    variant="link"
  >
    Invalid
  </Button>
</Popover>
`;

exports[`<SubscriptionStatusIndicator /> should render when subscription is NotSet 1`] = `
<Popover
  appendTo={[Function]}
  aria-label="Not Subscribed"
  bodyContent="This cluster does not have a subscription attached."
  className=""
  closeBtnAriaLabel="Close"
  enableFlip={true}
  footerContent={null}
  headerContent={null}
  hideOnOutsideClick={true}
  isVisible={null}
  maxWidth="calc(2rem + 18.75rem)"
  onHidden={[Function]}
  onHide={[Function]}
  onMount={[Function]}
  onShow={[Function]}
  onShown={[Function]}
  position="top"
  shouldClose={[Function]}
  zIndex={9999}
>
  <Button
    icon={
      <WarningTriangleIcon
        color="#f0ab00"
        noVerticalAlign={false}
        size="sm"
        title={null}
      />
    }
    isInline={true}
    variant="link"
  >
    Not Subscribed
  </Button>
</Popover>
`;

exports[`<SubscriptionStatusIndicator /> should render when subscription is Ok 1`] = `
<Fragment>
  <CheckCircleIcon
    color="#92d400"
    noVerticalAlign={false}
    size="sm"
    title={null}
  />
   
  Subscribed
</Fragment>
`;

exports[`<SubscriptionStatusIndicator /> should render when subscription is Overcommitted 1`] = `
<Popover
  appendTo={[Function]}
  aria-label="Insufficient"
  bodyContent="This cluster is consuming more resources than it is entitled to."
  className=""
  closeBtnAriaLabel="Close"
  enableFlip={true}
  footerContent={null}
  headerContent={null}
  hideOnOutsideClick={true}
  isVisible={null}
  maxWidth="calc(2rem + 18.75rem)"
  onHidden={[Function]}
  onHide={[Function]}
  onMount={[Function]}
  onShow={[Function]}
  onShown={[Function]}
  position="top"
  shouldClose={[Function]}
  zIndex={9999}
>
  <Button
    icon={
      <WarningTriangleIcon
        color="#f0ab00"
        noVerticalAlign={false}
        size="sm"
        title={null}
      />
    }
    isInline={true}
    variant="link"
  >
    Insufficient
  </Button>
</Popover>
`;

exports[`<SubscriptionStatusIndicator /> should render when subscription is Valid 1`] = `
<Fragment>
  <CheckCircleIcon
    color="#92d400"
    noVerticalAlign={false}
    size="sm"
    title={null}
  />
   
  Valid
</Fragment>
`;
